app/templates/components/billing-address.hbs

```diff
@@ -1,19 +1,243 @@
+                    @label="City"
+                    @value={{info.city}}
+                    @required={{true}}
+                    @validateOnField={{false}}
+                    @showValidationStatusIcons={{false}}
+                    @onChange={{action (mut info.city)}}
+                    as |field|
+                  >
+                    <field.input name="city" placeholder="City" />
+                  </form.field>
+                </div>
+                <div class="form-elem">
+                  <form.field
+                    @label="Post Code"
+                    @value={{info.zipCode}}
+                    @required={{true}}
+                    @validateOnField={{false}}
+                    @showValidationStatusIcons={{false}}
+                    @onChange={{action (mut info.zipCode)}}
+                    as |field|
+                  >
+                    <field.input name="postCode" placeholder="Post Code" />
+                  </form.field>
+                </div>  
+              </div>
+              <div class="form-elem">
+                {{#if this.editContact.isRunning}}
+                  <LoadingIndicator />
+                {{else}}
+                  <div class="flex flex--wrap">
+                    <button data-test-update-billing-address-button='true' onclick={{form.submit}} class="button--blue">
+                      Edit billing details
+                    </button>
+                    <a data-test-cancel-billing-address-button='true' onClick={{action (toggle 'openEditBillingForm' this)}} class='link cancel-link'>
+                      Cancel
+                    </a>
+                  </div>
+                {{/if}}
+              </div>
+            </TravisForm>
+          </div>
+        {{else}}
+          <div data-test-billing-details='true' class="flex flex--space-between flex--wrap">
+            <section>
+              {{#if info.address}}
+                <h4>address</h4>
+                <p>
+                  {{info.address}}
+                </p>
+              {{/if}}
+            </section>
+            <section>
+              {{#if info.city}}
+                <h4>city,state/territory</h4>
+                <p>{{info.city}}</p>
+              {{/if}}
+            </section>
+            <section>
+              <h4>post code</h4>
+              {{#if info.zipCode}}
+                <p>{{info.zipCode}}</p>
+              {{/if}}
+            </section>
+            <section>
+              {{#if info.country}}
+              <h4>country</h4>
+              <p>{{info.country}}</p>
+              {{/if}}
+            </section>
+          </div>
+          <button class='button--blue billing-subscription__buttons' 
+            onClick={{action (toggle 'openEditBillingForm' this)}} data-test-edit-billing-address='true'
+          >
+            Edit billing details
+          </button>
+        {{/if}}
+      </div>
+    {{/let}}
+  </section>
+{{/if}}
```

Comment:
This template is huge! I'd suggest to extract forms to different components, thus breaking the template to multiple smaller ones. What do you think?